lib/Service/SnoozeService.php

```diff
@@ -81,6 +83,79 @@ public function wakeMessages(): void {
 		}
 	}
 
+	/**
+	 * @param Message $message
+	 * @param int $unixTimestamp
+	 * @param Account $srcAccount
+	 * @param Mailbox $srcMailbox
+	 * @param Account $dstAccount
+	 * @param Mailbox $dstMailbox
+	 *
+	 * @return void
+	 *
+	 * @throws ServiceException
+	 */
+	public function snoozeMessage(
+		Message $message,
+		int $unixTimestamp,
+		Account $srcAccount,
+		Mailbox $srcMailbox,
+		Account $dstAccount,
+		Mailbox $dstMailbox
+	): void {
+		$this->snoozeMessageDB($message, $unixTimestamp);
+
+		try {
+			$this->mailManager->moveMessage(
+				$srcAccount,
+				$srcMailbox->getName(),
+				$message->getUid(),
+				$dstAccount,
+				$dstMailbox->getName()
+			);
+		} catch (\Throwable $e) {
+			$this->unSnoozeMessageDB($message);
+			throw new ServiceException($e->getMessage());
```

Comment:
loses all of the previous error trace. either just rethrow the original throwable or pass the original as *previous* of the new service exception

Reply:
:white_check_mark: